File: Go/ExampleTests/common/newpost_string.go
package common

type NewPostString struct {
	Title string
	Body string
	UserId string
}

func NewNewPostStringFromSlice(v []string) NewPostString {
	s := NewPostString{}
	if len(v) > 0 { s.Title = v[0] }
	if len(v) > 1 { s.Body = v[1] }
	if len(v) > 2 { s.UserId = v[2] }
	return s
}

// NewNewPostStringFromText builds from the text form, e.g. Money as "25 USD".
func NewNewPostStringFromText(text string) NewPostString {
	parts := RequireTokens(text, 3, "NewPost")
	return NewPostString{
		Title: parts[0],
		Body: parts[1],
		UserId: parts[2],
	}
}

func (s NewPostString) String() string {
	return Token(s.Title) + " " + Token(s.Body) + " " + Token(s.UserId)
}

func (s NewPostString) Equals(o NewPostString) bool {
	return DNCEqual(s.Title, o.Title) &&
		DNCEqual(s.Body, o.Body) &&
		DNCEqual(s.UserId, o.UserId)
}

func EqualNewPostStringSlices(a, b []NewPostString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/newpost_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct NewPostString {
    pub title: String,
    pub body: String,
    pub userid: String,
}

impl NewPostString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            title: v.get(0).copied().unwrap_or("").to_string(),
            body: v.get(1).copied().unwrap_or("").to_string(),
            userid: v.get(2).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 3, "NewPost");
        Self {
            title: parts[0].clone(),
            body: parts[1].clone(),
            userid: parts[2].clone(),
        }
    }
}

impl std::fmt::Display for NewPostString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.title), crate::common::tokens::token(&self.body), crate::common::tokens::token(&self.userid)].join(" "))
    }
}

impl PartialEq for NewPostString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.title, &other.title)
            && crate::common::dnc_equal(&self.body, &other.body)
            && crate::common::dnc_equal(&self.userid, &other.userid)
    }
}
